examples/summarization/bart/run_bart_sum.py

```diff
@@ -0,0 +1,211 @@
+        )
+        dataloader = DataLoader(train_dataset, batch_size=self.hparams.train_batch_size)
+        t_total = (
+            (len(dataloader.dataset) // (self.hparams.train_batch_size * max(1, self.hparams.n_gpu)))
+            // self.hparams.gradient_accumulation_steps
+            * float(self.hparams.num_train_epochs)
+        )
+        scheduler = get_linear_schedule_with_warmup(
+            self.opt, num_warmup_steps=self.hparams.warmup_steps, num_training_steps=t_total
+        )
+        self.lr_scheduler = scheduler
+        return dataloader
+
+    def val_dataloader(self):
+        val_dataset = CnnDailyMailDataset(
+            self.tokenizer, data_dir=self.hparams.data_dir, type_path="val", block_size=self.hparams.max_seq_length
+        )
+        return DataLoader(val_dataset, batch_size=self.hparams.eval_batch_size)
+
+    def test_dataloader(self):
+        test_dataset = CnnDailyMailDataset(
+            self.tokenizer, data_dir=self.hparams.data_dir, type_path="test", block_size=self.hparams.max_seq_length
+        )
+        return DataLoader(test_dataset, batch_size=self.hparams.eval_batch_size)
+
+    @staticmethod
+    def add_model_specific_args(parser, root_dir):
+        # Add BART specific options
+        parser.add_argument(
+            "--model_type", default="bart", type=str, help="Bart model type",
+        )
+        parser.add_argument(
+            "--max_seq_length",
+            default=1024,
+            type=int,
+            help="The maximum total input sequence length after tokenization. Sequences longer "
+            "than this will be truncated, sequences shorter will be padded.",
+        )
+
+        parser.add_argument(
+            "--model_name_or_path",
+            type=str,
+            default="bart-large",
+            help="Path to pre-trained model or shortcut name for BART",
+        )
+        parser.add_argument(
+            "--config_name", default="", type=str, help="Pretrained config name or path if not the same as model_name"
+        )
+        parser.add_argument(
+            "--do_lower_case", action="store_true", help="Set this flag if you are using an uncased model."
+        )
+        parser.add_argument(
+            "--tokenizer_name",
+            default="",
+            type=str,
+            help="Pretrained tokenizer name or path if not the same as model_name",
+        )
+        parser.add_argument(
+            "--cache_dir",
+            default="",
+            type=str,
+            help="Where do you want to store the pre-trained models downloaded from s3",
+        )
+
+        parser.add_argument(
+            "--data_dir",
+            default=None,
+            type=str,
+            required=True,
+            help="The input data dir. Should contain the training files for the CNN/DM summarization task.",
+        )
+
+        parser.add_argument("--learning_rate", default=3e-5, type=float, help="The initial learning rate for Adam.")
```

Comment:
Don't think you need these guys anymore

Reply:
We do otherwise we get `AttributeError: 'Namespace' object has no attribute 'learning_rate'` since these are model specific args and aren't part of `add_generic_args`

Reply:
Oh, you should just call the parent `add_model_specific_args` function before adding the BART args
https://github.com/huggingface/transformers/blob/master/examples/transformer_base.py#L148

They are mostly all here:
https://github.com/huggingface/transformers/blob/master/examples/transformer_base.py#L181